notes/stv/20210628-multi-user-deploy-01.txt

```diff
@@ -0,0 +1,153 @@
+        --volume "${SSH_AUTH_SOCK}:/mnt/ssh_auth_sock:rw,z" \
+        --volume "${HOME:?}/clouds.yaml:/etc/openstack/clouds.yaml:ro,z" \
+        --volume "${AGLAIS_CODE:?}/deployments:/deployments:ro,z" \
+        atolmis/ansible-client:2020.12.02 \
+        bash
+
+
+# -----------------------------------------------------
+# Set the target cloud to delete.
+#[root@ansibler]
+
+    cloudname=gaia-test
+
+
+
+# -----------------------------------------------------
+# Delete everything.
+#[root@ansibler]
+
+    time \
+        /deployments/openstack/bin/delete-all.sh \
+            "${cloudname:?}"
+
+
+	> Done
+
+# -----------------------------------------------------
+# Create everything, using a standard config.
+#[root@ansibler]
+
+    time \
+        /deployments/hadoop-yarn/bin/create-all.sh \
+            "${cloudname:?}" \
+            'medium-04'
+
+
+	> Done
+
+
+
+# ----------------------------------------------------------------
+# Login via Firefox, in two separate windows, as two separate users
+#[user@desktop]
+
+    firefox --new-window "http://128.232.227.136:8080/" &
+
+
+
+# -------------------------------------------------------
+# Run example notebooks, concurrently using both accounts
+#[user@desktop]
+
+ 
+    # Run imported notebooks from Zeppelin as user #1
+
+     /AglaisPublicExamples/SetUp                                                          
+        https://raw.githubusercontent.com/wfau/aglais-notebooks/main/2G7GZKWUH/note.json
+
+        [Success]
+   
+
+     /AglaisPublicExamples/Good astrometric solutions via ML Random Forrest classifier   
+        https://raw.githubusercontent.com/wfau/aglais-notebooks/main/2G5NU6HTK/note.json
+	 
+	Raw catalogue with selected columns
+	1724028
+	FINISHED   
+	Took 9 min 12 sec. Last updated by admin at July 01 2021, 11:21:26 AM.
+
+
+	Train up the Random Forrest
+	FINISHED
+	Took 5 min 53 sec. Last updated by gaiauser at July 01 2021, 11:27:34 AM.
+
+        [Success]
+  
+
+
+ 
+    # Run imported notebooks from Zeppelin as user #2, while application for user #1 is still running
+
+     /AglaisPublicExamples/SetUp                                                          
+        https://raw.githubusercontent.com/wfau/aglais-notebooks/main/2G7GZKWUH/note.json
+
+
+	Raw catalogue with selected columns
+	1724028
+	FINISHED
+	Took 9 min 24 sec. Last updated by gaiauser at July 01 2021, 11:21:26 AM.
+
+	
+	Train up the Random Forrest
+        FINISHED   
+        Took 5 min 50 sec. Last updated by admin at July 01 2021, 11:27:30 AM.
+
+        [Success]
+
+
+
+     /AglaisPublicExamples/Good astrometric solutions via ML Random Forrest classifier   
+        https://raw.githubusercontent.com/wfau/aglais-notebooks/main/2G5NU6HTK/note.json
+	
+        [Success]
+  
+
+
+# While observing the resource usage using the Spark UI, we see two applications running, each with 47.1% usage
+
+# Also worth noting that both notebooks took about the same time to complete
```

Comment:
:+1: good notes explain what is happening and how we arrived at this result.

Reply:
It would be good to have a single user run using the same configuration to act as a comparison.

* user A only - Took 5 min 50 sec
* user B only - Took 5 min 50 sec
* user A and B concurrently - Took 5 min 50 sec

Reply:
Can we run the same set of tests from the command line using curl ?
That would make the tests much more reproducible.

Reply:
This is not a shop stopper - we can merge without this.

Reply:
OK - not a show stopper, but without this comparison your statement that _"both notebooks took about the same time to complete"_ doesn't have enough evidence.